apps/cli/src/commands/coprocessor/deploy.ts

```diff
@@ -0,0 +1,490 @@
+import { Command } from "@commander-js/extra-typings";
+import chalk from "chalk";
+import { execa } from "execa";
+import fs from "fs-extra";
+import ora from "ora";
+import path from "path";
+import { createWalletClient, http } from "viem";
+import { privateKeyToAccount } from "viem/accounts";
+import { waitForTransactionReceipt } from "viem/actions";
+import { foundry } from "viem/chains";
+import { getMachineHashFromDir } from "../../base.js";
+
+export const createDeployCommand = () => {
+    return new Command("deploy")
+        .description("Deploys a coprocessor framework application.")
+        .argument(
+            "<contractName>",
+            "Name of solidity contract you intend to deploy",
+        )
+        .option(
+            "-c, --constructorArgs [args...]",
+            "Optional list of constructor arguments for the contract (space-separated values)",
+        )
+        .configureHelp({ showGlobalOptions: true })
+        .action(async (contractName: string, options) => {
+            try {
+                let constructorArgs: string[] | [] = [];
+                if (
+                    options.constructorArgs == true ||
+                    options.constructorArgs == undefined
+                ) {
+                    constructorArgs = [];
+                } else {
+                    constructorArgs = options.constructorArgs;
+                }
+
+                let { cartesiDir, foundryDir } = await findSubdirectories();
+                if (cartesiDir && foundryDir) {
```

Comment:
Code structure is too complex.
Should be something like:

```typescript
const car = await pack({ ... });
await publish({ car ... });
await deploy({ ... });
```

exceptions raised by any step will stop execution and show the error.